stingray/varenergyspectrum.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import division
+import numpy as np
+from stingray.gti import check_separate, cross_two_gtis
+from stingray.lightcurve import Lightcurve
+from stingray.utils import assign_value_if_none, simon
+from stingray.crossspectrum import AveragedCrossspectrum
+import six
+
+
+class VarEnergySpectrum(object):
+    def __init__(self, events, freq_interval, energy_spec, ref_band,
+                 bin_time=1, use_pi=False, log_distr=False,
+                 segment_size=None, events2=None):
+
+        """Generic variability-energy spectrum.
+        
+        Parameters
+        ----------
+        events : stingray.events.EventList object 
+            event list
+        freq_interval : [f0, f1], floats
+            the frequency range over which calculating the variability quantity
+        energy_spec : [emin, emax, N]
```

Comment:
Yes, I was unsure about this too. But I also like giving a simple way to produce meaningful intervals. What about doing something like: if you give three numbers and the last is an integer, I produce equally-spaced (or logarithmically-spaced) intervals; if you provide an array with shape (N, 2), I use those as intervals?